### DISCUSSION TITLE
Is react-xr applicable for React Native projects?

### DISCUSSION THREAD

**Discussion by dbezrukov** [pmndrs/xr#156] [Q&A] [answered]
Hi, 
thank you for such an inspiring project!

At Customuse we're working on a React Native mobile app that allows configuring 3d outfits using React Three Fiber and Pixi. Actually we have two implementations - using imperative Expo Three approach and using declarative Expo Three Fiber one. Each outfit is represented as a gltf model with customisable texture.

The next step we are trying to do is to enable AR for viewing outfits.
I had hoped that replacing Canvas with ARCanvas would do the trick but ARCanvas tries to access document.querySelector which means that the component is designed for WebXR only. 

Could you please tell me will r3f be extended to both Web and Mobile platforms?
And what is the best option to enable AR (with face and, preferably, body tracking) in React Three Fiber powered mobile app? Can react-xr help to achieve this or it's better to look towards native iOS solutions like USBZ/RealityKit?

**Comment by CodyJasonBennett** [comment, marked as answer]
R3F v8 brings support for react-native, but native AR features are very limited on mobile devices (not implemented; also see [caniuse](https://caniuse.com/webxr)). Unfortunately, I don't believe Apple intends to implement this any time soon until they can better integrate the spec with their ecosystem.

I'd look for any native solutions you can that are cross-platform (IOS & Android) as AR isn't feasible to implement on your own. For tracking, you can find many Tensorflow projects with object/OpenCV & pose tracking as a start.

**Reply by r4topunk** [reply]
Any update here? I would like to build native applications with ReactXR, I'm hating the PWA workflow

**Reply by bbohlender** [reply]
People are working on this, but this work is not related to this repository.
Once there is a solution to bring WebXR apps to native, we will definitely mention it here.

**Comment by tomsoderlund** [comment]
@CodyJasonBennett Would it be possible to build something on iOS with `react-xr` and https://github.com/expo/expo-three-ar ?

**Reply by CodyJasonBennett** [reply]
Yes, but it's a lot of work. You could polyfill WebXR over JSI if you really wanted to. Expo depreciated their AR projects BTW, you'd be talking to ARCore on iOS.

**Reply by iShelar** [reply]
@tomsoderlund did you try creating the project?
